v2/auth/cachedauth/sign_in.go

```diff
@@ -50,16 +62,16 @@ func SignIn(ctx context.Context, username, password string) (err error) {
 		return errors.New("cachedauth is not configured")
 	}
 
+	oldTokens := tokens[username]
 	const tokenExpireDurationDiff = 5 * time.Minute
-	if auth.IsTokenValid(tokens.AccessToken, tokenExpireDurationDiff) {
+
+	if auth.IsTokenValid(oldTokens.AccessToken, tokenExpireDurationDiff) {
 		return nil
 	}
 
-	tokens, err = auth.SignIn(ctx, username, password)
-	if err != nil {
-		tokens = auth.Tokens{}
-		return err
-	}
+	newtokens, err := auth.SignIn(ctx, username, password)
+	tokens[username] = newtokens
```

Comment:
Will newtokens be empty on error? The old code did not assume this?

Reply:
Yes, that was the case in the old code aswell it is stated explicitly on line 60 in the old code